crates/env/src/engine/e2e/xts.rs

```diff
@@ -0,0 +1,360 @@
+                    "error on call `wait_for_in_block`: {:?}",
+                    err
+                );
+            })
+            .fetch_events()
+            .await
+            .unwrap_or_else(|err| {
+                panic!(
+                    "error on call `fetch_events`: {:?}",
+                    err
+                );
+            })
+    }
+
+    /// Dry runs a call of the contract at `contract` with the given parameters.
+    pub async fn call_dry_run(
+        &self,
+        contract: C::AccountId,
+        value: E::Balance,
+        storage_deposit_limit: Option<E::Balance>,
+        data: Vec<u8>,
+    ) -> ContractDryCallResult<E> {
+        let call_request = RpcCallRequest::<C> {
+            origin: contract.clone(),
+            dest: contract,
+            value: value.try_into().expect("unable to convert `value`"),
+            gas_limit: DRY_RUN_GAS_LIMIT.into(),
+            storage_deposit_limit: storage_deposit_limit.map(|l| {
+                l.try_into()
+                    .expect("unable to convert `storage_deposit_limit`")
+            }),
+            input_data: Bytes(data),
+        };
+        let params = rpc_params![call_request];
+        self.ws_client
+            .request("contracts_call", params)
+            .await
+            .unwrap_or_else(|err| {
+                panic!("error on ws request `contracts_call`: {:?}", err);
+            })
+    }
+
+    /// Submits an extrinsic to call a contract with the given parameters.
+    ///
+    /// Returns when the transaction is included in a block. The return value
+    /// contains all events that are associated with this transaction.
+    pub async fn call(
+        &self,
+        contract: sp_runtime::MultiAddress<C::AccountId, ()>,
+        value: E::Balance,
+        gas_limit: crate::types::Gas,
+        storage_deposit_limit: Option<E::Balance>,
+        data: Vec<u8>,
+        signer: &Signer<C>,
+    ) -> TxEvents<C> {
+        let call = subxt::tx::StaticTxPayload::new(
+            "Contracts",
+            "call",
+            Call::<C, E::Balance> {
+                dest: contract,
+                value,
+                gas_limit,
+                storage_deposit_limit,
+                data,
+            },
+            Default::default(),
+        )
+        .unvalidated();
+
+        self
+            .client
+            .tx()
+            .sign_and_submit_then_watch_default(&call, signer)
+            .await
+            .map(|tx_progress| {
+                log_info(&format!("signed and submitted call with extrinsic hash {:?}", tx_progress.extrinsic_hash()));
+                tx_progress
+            })
+            .unwrap_or_else(|err| {
+                panic!(
+                    "error on call `sign_and_submit_then_watch_default`: {:?}",
+                    err
+                );
+            })
+            // TODO(#xxx) It should be configurable to use `.wait_for_finalized_success` instead.
```

Comment:
Since this is for testing I don't think waiting for finalization is necessary?

Reply:
Hmm, so I agree, but I'm not sure that in the future if we merge XCM into ink! and have the e2e for it, that cross-chain testing will work without waiting for finalization. But I guess we can figure it out then!

Reply:
Yes let's not worsen the status quo for some future stuff.